### PR TITLE
feat: Add Link UX/UI improvements

### LINKED ISSUE #465: UX/UI updates to the "Add link" flow
New flow: https://www.figma.com/file/iPOdDkRxCaeoDoQ4JXjxlk/Space-Force-Service-Portal?node-id=3228%3A31669

Changes:
- Modify the Add Custom Link modal to include text fields for both Name & URL (both required, URL must be valid)
- When typing in the Add Link ComboBox, display the "Add Custom Link" option if no results are found. Selecting this option opens the modal.
- Display the Custom Link Icon on custom bookmarks, with an edit icon instead of delete
- Clicking the edit icon opens the Custom Link modal with a delete button. User can modify the URL and/or name of the link here.

### PR DESCRIPTION
**Pull request by suzubara**
## Description 

This PR makes the following enhancements to the Add Link flow:

* Custom links in a collection get a different UI treatment
* The Add Link ComboBox input only allows selecting an existing CMS link
* Adding a custom link is handled separately, in a modal
* Users can edit & delete custom links via modal

Also adds Storybook pages for all existing modal components.

Fixes #465 

## Review Notes

image

image

image

image

### CODE REVIEW

File: src/cms/schema.test.ts [hunk 1]
```diff
@@ -1,7 +1,18 @@
 import { setupTestRunner } from '@keystone-next/keystone/testing'
-import config from '../../keystone'
+import { config } from '@keystone-next/keystone'
 
-const runner = setupTestRunner({ config })
+import schema from './schema'
+
+const testConfig = config({
+  db: { provider: 'sqlite', url: 'file:./test-cms.db' },
```
Comment (suzubara) [inline src/cms/schema.test.ts:7]: @abbyoung this should fix the CMS content getting wiped out locally!

Comment (abbyoung) [reply src/cms/schema.test.ts:7]: EXCELLENT!

File: src/components/MySpace/MySpace.test.tsx [hunk 2]
```diff
@@ -320,7 +319,15 @@ describe('My Space Component', () => {
...
     )
-    userEvent.click(screen.getByRole('button', { name: 'Delete' }))
+
+    const confirmDelete = screen.getByRole('dialog', {
+      name: 'Are you sure you’d like to delete this collection from My Space?',
+    })
+
+    expect(confirmDelete).toBeVisible()
+    userEvent.click(
+      within(confirmDelete).getByRole('button', { name: 'Delete' })
```
Comment (abbyoung) [inline src/components/MySpace/MySpace.test.tsx:329]: I'll have to remember to use `within`!

File: src/components/CustomCollection/CustomCollection.tsx [hunk 3]
```diff
@@ -280,12 +229,29 @@ const CustomCollection = ({
           dropdownRef={dropdownEl}
           align="right"
           isActive={isDropdownOpen}>
-          {customCollectionDropdownItems}
+          <ol>
```
Comment (abbyoung) [inline src/components/CustomCollection/CustomCollection.tsx:232]: thanks for all the extra comments and organizing in this file ⭐

**Review by abbyoung — DISMISSED** [review]
🙌

File: e2e/cypress/integration/sites-and-applications.spec.js [hunk 4]
```diff
@@ -241,67 +233,116 @@ describe('Sites and Applications', () => {
...
+    cy.findByRole('dialog', { name: 'Edit custom link' }).within(() => {
+      cy.findByRole('button', { name: 'Delete' }).click()
+    })
+
+    cy.findByRole('link', {
+      name: 'Edited Custom Link (opens in a new window)',
+    }).should('not.exist')
+    cy.contains('Example Collection')
+      .parent()
+      .parent()
```
Comment (abbyoung) [inline e2e/cypress/integration/sites-and-applications.spec.js:340]: I think you'll need an additional `.parent()` now to have the cypress test pass

Comment (suzubara) [reply e2e/cypress/integration/sites-and-applications.spec.js:344]: thank you!!

**Review by jbecker01 — CHANGES_REQUESTED** [review]
Add ellipses for longer names in order to preserve the custom icon. Currently, the custom icon is being removed when the URL name is too long. 
Screen Shot 2022-02-01 at 4 35 35 PM

Can we also add placeholder text to the Add a custom link fields:
Name field placeholder text: Example title name
URL field placeholder text: https://www.copy-paste-your-url.com

**Comment by suzubara** [comment]
@jbecker01 nice catch! This should be fixed now: 
image

Also, any consensus on placeholder text for the modal?

**Comment by suzubara** [comment]
Also added placeholder text! "Example title name" seemed like a typo, I made it "Example link name"

image

**Review by jbecker01 — APPROVED** [review]
Looks good on Chrome, FF, Safari, and Edge

I was unable to test on IE11 as I couldn't make it past our custom loader
loading

**Review by abbyoung — APPROVED** [review]

**Comment by suzubara** [comment]
> Looks good on Chrome, FF, Safari, and Edge
> 
> I was unable to test on IE11 as I couldn't make it past our custom loader loading

Follow up issue has been created here: https://github.com/USSF-ORBIT/ussf-portal-client/issues/505

**Pull request merged**